### PR TITLE
ossia-score: init at 3.1.11

### PR DESCRIPTION
**Pull request by minijackson**
###### Description of changes

A really cool application for audio / video / etc.: https://ossia.io/score/about.html

I don't have my whole setup on hand right now so I didn't do extensive checks with using vst(3)/lv2 plugins, but I'm putting it out there in case someone is interested.

Pinging the previously maintainer of `i-score`, the ancestor of ossia-score (sorry if you are not interested): @magnetophon 

On a side note, I'd be really down to create an "audio-production" nixpkgs workgroup.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jcelerier** [comment]
hey, ossia dev here, thanks for doing this ! if there's anything I can do to help ensuring a smooth packaging experience please tell me

**Comment by jcelerier** [comment]
also the whole ossia release workflow is centered on automating everything from gh actions ; one thing I've been thinking about more and more is automating creation of PRs, etc for updating things automatically whenever a new release is tagged.. if anyone has experience with this ?

**Comment by magnetophon** [comment]
@jcelerier Thanks for showing up here and offering support!

@minijackson Sorry for the late reply,
I'd be up for  an "audio-production" nixpkgs workgroup.

File: pkgs/applications/audio/ossia-score/default.nix
Comment (ctem) [inline pkgs/applications/audio/ossia-score/default.nix:34]:
```suggestion
, suil-qt5
```

File: pkgs/applications/audio/ossia-score/default.nix
Comment (ctem) [inline pkgs/applications/audio/ossia-score/default.nix:81]:
```suggestion
    suil-qt5
```

File: pkgs/applications/audio/ossia-score/default.nix
Comment (ctem) [inline pkgs/applications/audio/ossia-score/default.nix:91]:
```suggestion
    "-DSuil_INCLUDE_DIR=${suil-qt5}/include/suil-0"
```

File: pkgs/applications/audio/ossia-score/default.nix
Comment (ctem) [inline pkgs/applications/audio/ossia-score/default.nix:92]:
```suggestion
    "-DSuil_LIBRARY=${suil-qt5}/lib/libsuil-0.so"
```

**Review by ctem — CHANGES_REQUESTED** [review]
The `suil` package is for the Qt 4 version, but this PR builds score against Qt 5. Without this change, score will still build and run, but custom UI's for LV2 plugins won’t load, instead throwing the following error:

```
suil error: Unable to wrap UI type <http://lv2plug.in/ns/extensions/ui#X11UI> as type <http://lv2plug.in/ns/extensions/ui#Qt5UI>
```

**Comment by ctem** [comment]
Thanks @minijackson for your work so far on this. As @jcelerier and I discussed in the score Matrix room, this PR builds and runs both as-is and for version 3.1.1. (I think the build fails for 3.1.2 but haven’t investigated.) I finally figured out the problem with the custom UI display of LV2 plugins (please see the review).

I’m with @magnetophon in being interested in an audio production nixpkgs workgroup. Perhaps the [NixOS Audio Matrix room](https://matrix.to/#/#audio:nixos.org) is a good place to discuss.

**Comment by magnetophon** [comment]
@ctem I have Joined the matrix room, thanks!
@minijackson Would matrix work for you as well?

**Comment by jcelerier** [comment]
thanks for finding the issue ! wouldn't have been able to. Can you share a build log ? 
Also if there's a way to add a Nix CI to github actions I'll add it to score's repo so that we have some info on it

**Comment by minijackson** [comment]
Thanks a lot everyone. I'm a bit busy at the moment, with NixCon coming up and some personal stuff, but I really do appreciate a lot the discussions here.

@ctem @magnetophon I'll be seeing you on the Matrix side, then!

@jcelerier For automated PRs, we already have [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) which works pretty well. You can see [PRs by r-ryantm](https://github.com/NixOS/nixpkgs/pulls/r-ryantm). Thanks a lot for offering your help :heart: 

In any case, I'll integrate the review, and investigate 3.1.2, soon™

**Comment by pjhenning** [comment]
Hello, I've been hacking on a variant of this package definition to get a working build of `master` running on aarch64 and with ysfx integration, would the aforementioned matrix room be a good place to discuss if/how I can contribute to this PR, or is it better to do so here via comments?

**Comment by giacomocariello** [comment]
Any update on this?

**Comment by jcelerier** [comment]
one relevant thing I can tell anyone maintaining this package: score now has migrated to and exclusively uses Qt 6 (ideally 6.5+ due to some relevant fixes that made it in Qt) so recips may need updating

**Comment by luzpaz** [comment]
soft bump

**Comment by luzpaz** [comment]
bump

File: pkgs/applications/audio/ossia-score/default.nix
Comment (minijackson) [reply pkgs/applications/audio/ossia-score/default.nix:34]: I'm not seeing the `suil-qt5` package anymore. Was it removed?

**Comment by minijackson** [comment]
Sorry for the very late update.

I've got some good and bad news:

the good news is that I've updated the PR, and it builds and runs

the bad news is that the latest released ossia-score doesn't support Qt 6.6, so I've pushed a build from master, but we'll need to wait until next release.

Note: I had a compilation issue that I've bypassed by enabling the unity build: `/build/source/src/plugins/score-plugin-gfx/Gfx/Filter/PreviewWidget.hpp:20:36: error: invalid use of incomplete type 'class QWidget'`. @jcelerier do you want me to create an issue?

I also have got very bad graphic card issues on my computer, so I cannot properly test this: there's lots of graphical issues that are probably coming from my end. I'll really appreciate it if someone else could test this PR.

Also, if someone wants to co-maintain ossia-score in nixpkgs, don't hesitate!

**Comment by magnetophon** [comment]
Result of `nixpkgs-review pr 174802` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ossia-score</li>
  </ul>
</details>

It runs, the web links in the startup menu work and I was able to add an audio object.
Other than that, I have no idea how it works and how to test it.

**Comment by jcelerier** [comment]
> the bad news is that the latest released ossia-score doesn't support Qt 6.6, so I've pushed a build from master, but we'll need to wait until next release.

ah yes, sorry I'm super late for tagging a release but current master is tested with 6.6 (and that's what next release will build against too, I just hit some deep problems with symbol limits on Win32 which cause me to have to rethink the entire build...)

**Comment by jcelerier** [comment]
updated to 3.1.12

**Comment by luzpaz** [comment]
@jcelerier any progress ?

**Comment by luzpaz** [comment]
Any progress here ?

**Comment by minijackson** [comment]
PR should be ready now.

If anyone wants to co-maintain ossia-score, I'd be more than happy to add you to the maintainer list!

**Comment by magnetophon** [comment]
Result of `nixpkgs-review pr 174802` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ossia-score</li>
  </ul>
</details>

It also runs.

**Comment by jcelerier** [comment]
Saw a small aarch64 build failure, I committed already, will be on the next release

**Comment by luzpaz** [comment]
3.1.13 has been released (https://github.com/ossia/score/releases/tag/v3.1.13)

**Comment by jcelerier** [comment]
we're at 3.2.0 now

**Comment by jcelerier** [comment]
I made a new PR to update to the current version: https://github.com/NixOS/nixpkgs/pull/315409

**Comment by jcelerier** [comment]
I think this can be closed

**Pull request closed without merging**
